pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -351,6 +351,7 @@ public Future<Void> sendMessages(final List<? extends Entry> entries, EntryBatch
                    topicName, subscription, ackedCount, totalMessages, consumerId, avgMessagesPerEntry.get());
         }
         incrementUnackedMessages(unackedMessages);
+        final int finalUnackedMessages = unackedMessages;
```

Comment:
Suggest to add the logic in the `incrementUnackedMessages`.

Reply:
> Suggest to add the logic in the `incrementUnackedMessages`.

Good suggestion, thanks.